lib/internal/modules/esm/resolve.js

```diff
@@ -43,6 +45,16 @@ const {
   ERR_UNSUPPORTED_ESM_URL_SCHEME,
 } = require('internal/errors').codes;
 
+const DEFAULT_CONDITIONS = ObjectFreeze(['node', 'import']);
+const DEFAULT_CONDITIONS_SET = new SafeSet(DEFAULT_CONDITIONS);
+
+function getConditionsSet(conditions) {
+  if (conditions !== DEFAULT_CONDITIONS && ArrayIsArray(conditions)) {
```

Comment:
We should throw an error ERR_INVALID_ARG_TYPE for the loader if it is not an array or undefined / null.

Reply:
Should we validate that all items are strings? I think default coercion to string is probably fine for all existing types / cannot imagine non-string items being useful right now.

Reply:
I don't think coercion is happening right now (if I understand your assumption correctly). Any non-string element in the array should be ignored right now since we're using set membership tests.

Reply:
I added a check that throws when it's a non-array.